src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js

```diff
@@ -43,29 +44,50 @@ const defaultViewProps = {style: [styles.alignItemsStart, styles.userSelectText]
 // context to RenderHTMLSource components. See https://git.io/JRcZb
 // Beware that each prop should be referentialy stable between renders to avoid
 // costly invalidations and commits.
-const BaseHTMLEngineProvider = (props) => {
-    // We need to memoize this prop to make it referentially stable.
-    const defaultTextProps = useMemo(() => ({selectable: props.textSelectable}), [props.textSelectable]);
+class BaseHTMLEngineProvider extends React.Component {
+    constructor(props) {
+        super(props);
 
-    return (
-        <TRenderEngineProvider
-            customHTMLElementModels={customHTMLElementModels}
-            baseStyle={styles.webViewStyles.baseFontStyle}
-            tagsStyles={styles.webViewStyles.tagStyles}
-            enableCSSInlineProcessing={false}
-            systemFonts={_.values(fontFamily)}
-        >
-            <RenderHTMLConfigProvider
-                defaultTextProps={defaultTextProps}
-                defaultViewProps={defaultViewProps}
-                renderers={htmlRenderers}
-                computeEmbeddedMaxWidth={HTMLEngineUtils.computeEmbeddedMaxWidth}
+        this.appStateChangeListener = null;
+        this.state = {
+            pixelRatio: PixelRatio.getFontScale(),
+        };
+    }
+
+    componentDidMount() {
+        this.appStateChangeListener = AppState.addEventListener('change', () => {
+            this.setState({pixelRatio: PixelRatio.getFontScale()});
+        });
+    }
+
+    componentWillUnmount() {
+        if (!this.appStateChangeListener) { return; }
+        this.appStateChangeListener.remove();
+    }
+
+    render() {
+        const defaultTextProps = {selectable: this.props.textSelectable};
+        return (
+            <TRenderEngineProvider
+                customHTMLElementModels={customHTMLElementModels}
+                baseStyle={styles.webViewStyles(this.state.pixelRatio).baseFontStyle}
+                tagsStyles={styles.webViewStyles(this.state.pixelRatio).tagStyles}
+                enableCSSInlineProcessing={false}
+                dangerouslyDisableWhitespaceCollapsing
```

Comment:
Not sure if I understand or am comfortable with this change. What is dangerous about it?

Reply:
we converted `webViewStyles` object in `styles.js` to a function return an object because we need to change the `fontSize` inside this object from `BaseHTMLEngineProvider.js` at any time the user try to change the device font size while the App is opening, so we call this function using the new `pixelRatio` value and get the style object with the new `fontSize` value.

Reply:
It seems you mean this line`dangerouslyDisableWhitespaceCollapsing`, I didn't add it, it was in [1.1.54-1 Release](https://github.com/Expensify/App/releases/tag/1.1.54-1) and removed 17 days ago in [this commit](https://github.com/Expensify/App/pull/9025/commits/09857bf11e74319a1904a522cc2f43f7e96690dc) after i forked the repo, i fetched the appstream/main multiple times before i don't know why it still there!!